lib.php

```diff
@@ -0,0 +1,59 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Libs.
+ * @copyright  Catalyst IT 2022
+ * @package    local_login
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+/**
+ * Triggers after site config is loaded. It is used to
+ *
+ */
+function local_login_after_config() {
+    global $CFG, $SESSION, $_SERVER, $USER, $DB;
+    $errorcode = 0;
+    $noredirect  = optional_param('noredirect', 0, PARAM_BOOL); // don't redirect
+    if (!isloggedin()) {
+        // Set new login URL.
+        $loginurl = new moodle_url('/local/login/index.php');
+        $loginurlstr = $loginurl->out(false);
+
+        // To stop redirect if alternateloginurl is set.
+        if (!empty($CFG->alternateloginurl)) {
+            unset($CFG->alternateloginurl);
+        }
+
+        if (!empty($SESSION->wantsurl) && strpos($SESSION->wantsurl, $loginurlstr) === 0) {
+            // We do not want to return to alternate url.
+            $SESSION->wantsurl = null;
+        } else if (isset($_SERVER['REQUEST_URI'])) {
```

Comment:
I have added the comments on this. It is needed, the first check prevents the ending redirect as wantsurl is login to local/login and the later  was added to resolve phpunittest failures like:
- vendor/bin/phpunit --testsuite workshopform_rubric_testsuite
- vendor/bin/phpunit --testsuite workshopform_numerrors_testsuite

Reply:
Also I had to add this line https://github.com/catalyst/moodle-local_login/blob/noredirect/index.php#L67 as otherwise if the user came from let's say a course view page then he was not able to login but instead kept redirected to the local/login page.

Reply:
Resetting wantsurl is losing the feature of redirecting a user to the page he was before logging in and might have other implications as well. I have explore using another way of achieving this.